test/integration/util/util.go

```diff
@@ -481,6 +520,16 @@ func InitTestAPIServer(t *testing.T, nsPrefix string, admission admission.Interf
 		},
 	})
 
+	// Support wrapping HTTP requests.
+	testCtx.KubeConfig.Wrap(func(transport http.RoundTripper) http.RoundTripper {
+		return roundTripWrapper{tc: testCtx, transport: transport}
+	})
+	var err error
+	testCtx.ClientSet, err = clientset.NewForConfig(testCtx.KubeConfig)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
nit: `CleanupTest(t, testCtx)` before this line?

Reply:
That's also problematic. Suppose initialization fails halfway through, for example before creating the namespace. Then when `CleanupTest` runs, it fails in `framework.DeleteNamespaceOrDie` and `testCtx.CloseFn` doesn't get called.

The right solution is to remove `CleanupTest` and replace it with individual cleanup functions that get registered if (and only if) needed. That's a bigger change that doesn't belong into this PR.